feat: add favourite icon toggler lab (#517)
freeCodeCamp/CurriculumExpansion: frontend-cert/js-projects/lab-favourite-toggler/user-stories.md

> You should have an `h1` heading that contains the a heading.

Reviewer: ```suggestion
1. You should have an `h1` element that contains the heading.
```

> You should have an unordered list with three items.
> The unordered list should have the class `item-list`.

Reviewer: ```suggestion
1. You should have an unordered list with three items.
1. The unordered list should have the class `item-list`.
```
Are these two separate user stories?

> The three list items should contain the item name followed by a `span` element with the class `favorite-icon ` and an id of `favorite-icon-1`, `favorite-icon-2`, and `favorite-icon-3` respectively.

Reviewer: ```suggestion
1. The three list items should contain the item name followed by a `span` element with the class `favorite-icon` and an id of `favorite-icon-1`, `favorite-icon-2`, and `favorite-icon-3`, respectively.
```

> The `span` element should contain the code `&#9825;` initially to represent an empty heart.

Reviewer: ```suggestion
1. The `span` element should contain the code `&#9825;` initially to represent an empty heart.
```

> When the `span` element is clicked, the class `filled` should be added to the `span` with class `favorite-icon` if it doesn't exist, and removed if it does.  `&#9825;` should change to `&#10084;` to represent a filled heart.

Reviewer: ```suggestion
1. When the `span` element is clicked, the class `filled` should be added to the `span` with class `favorite-icon` if it doesn't exist, and removed if it does.  `&#9825;` should change to `&#10084;` to represent a filled heart.
```
